Firestore/Example/Tests/Model/FSTMutationTests.mm

```diff
@@ -185,7 +185,116 @@ - (void)testCreateArrayRemoveTransform {
   }
 }
 
-- (void)testAppliesServerAckedTransformsToDocuments {
+- (void)testAppliesLocalArrayUnionTransformToMissingField {
+  auto baseDoc = @{};
+  auto transform = @{ @"missing" : [FIRFieldValue fieldValueForArrayUnion:@[ @1, @2 ]] };
+  auto expected = @{ @"missing" : @[ @1, @2 ] };
+  [self transformBaseDoc:baseDoc with:transform expecting:expected];
+}
+
+- (void)testAppliesLocalArrayUnionTransformToNonArrayField {
+  auto baseDoc = @{ @"non-array" : @42 };
+  auto transform = @{ @"non-array" : [FIRFieldValue fieldValueForArrayUnion:@[ @1, @2 ]] };
+  auto expected = @{ @"non-array" : @[ @1, @2 ] };
+  [self transformBaseDoc:baseDoc with:transform expecting:expected];
+}
+
+- (void)testAppliesLocalArrayUnionTransformWithNonExistingElements {
+  auto baseDoc = @{ @"array" : @[ @1, @3 ] };
+  auto transform = @{ @"array" : [FIRFieldValue fieldValueForArrayUnion:@[ @2, @4 ]] };
+  auto expected = @{ @"array" : @[ @1, @3, @2, @4 ] };
+  [self transformBaseDoc:baseDoc with:transform expecting:expected];
+}
+
+- (void)testAppliesLocalArrayUnionTransformWithExistingElements {
+  auto baseDoc = @{ @"array" : @[ @1, @3 ] };
+  auto transform = @{ @"array" : [FIRFieldValue fieldValueForArrayUnion:@[ @1, @3 ]] };
+  auto expected = @{ @"array" : @[ @1, @3 ] };
+  [self transformBaseDoc:baseDoc with:transform expecting:expected];
+}
+
+- (void)testAppliesLocalArrayUnionTransformWithDuplicateExistingElements {
+  // Duplicate entries in your existing array should be preserved.
+  auto baseDoc = @{ @"array" : @[ @1, @2, @2, @3 ] };
+  auto transform = @{ @"array" : [FIRFieldValue fieldValueForArrayUnion:@[ @2 ]] };
+  auto expected = @{ @"array" : @[ @1, @2, @2, @3 ] };
+  [self transformBaseDoc:baseDoc with:transform expecting:expected];
+}
+
+- (void)testAppliesLocalArrayUnionTransformWithDuplicateUnionElements {
+  // Duplicate entries in your union array should only be added once.
+  auto baseDoc = @{ @"array" : @[ @1, @3 ] };
+  auto transform = @{ @"array" : [FIRFieldValue fieldValueForArrayUnion:@[ @2, @2 ]] };
+  auto expected = @{ @"array" : @[ @1, @3, @2 ] };
+  [self transformBaseDoc:baseDoc with:transform expecting:expected];
+}
+
+- (void)testAppliesLocalArrayUnionTransformWithNonPrimitiveElements {
+  // Union nested object values (one existing, one not).
+  auto baseDoc = @{ @"array" : @[ @1, @{@"a" : @"b"} ] };
+  auto transform =
+      @{ @"array" : [FIRFieldValue fieldValueForArrayUnion:@[ @{@"a" : @"b"}, @{@"c" : @"d"} ]] };
```

Comment:
Is it worth a test to verify that we don't match partial matches?

Something like
```objc
auto baseDoc = @{ @"array" : @[ @1, @{@"a" : @"b", @"c": @"d" } ] };
auto transform =
     @{ @"array" : [FIRFieldValue fieldValueForArrayUnion:@[ @{@"a" : @"b"}, @{@"c" : @"d"} ]] };
auto expected = @{ @"array" : @[ @1, @{@"a" : @"b", @"c": @"d" }, @{@"a" : @"b"}, @{@"c" : @"d"} ] };
```